### PR TITLE
Sphinx 4 Support

### PR DESCRIPTION
**Pull request by Daltz333**

### CODE REVIEW

**Comment by Daltz333** [comment]
@foster999 could you take a look at the incompatibility with Sphinx 4?

**Comment by foster999** [comment]
Thanks for this @Daltz333 

I've compared outputs between sphinx 3.5.4 and 4 (pre-release). Looks like the formatting of `code-block`'s with line numbers has changed from:
image

to:

image

This seems to be an intentional change to improve accessibility (sphinx-doc/sphinx#7849), so I'll push a new sphinx-version-dependent test for this.

**Comment by Daltz333** [comment]
We may want to default html_codeblock_linenos_style to the old style for the user

**Comment by foster999** [comment]
Good point!

I've just tried setting the default to the old style in `sphinx-tabs`, but it seems like users are then unable to override this in their sphinx config.

Looking at the [sphinx changelog](https://github.com/sphinx-doc/sphinx/blob/4.x/CHANGES), they're depreciating the `html_codeblock_linenos_style` option too. Given this, and that the new format improves accessibility, I'd be more inclined to follow their lead. We could document that users should use the sphinx option to revert the style (or stick with sphinx 3) and/or bump to a new major version to reflect that sphinx 4 causes a non-backwards compatible change?

**Comment by Daltz333** [comment]
Yeah, didn't realize they are deprecating the "table" option

**Comment by Daltz333** [comment]
I've pushed my changes, but I'm not sure how to conditionally skip the regression test

**Comment by Daltz333** [comment]
Okay, got things figured out.

Duplicated the ``other_assets`` test to include the Sphinx 4 generated output. They are conditionally skipped depending on sphinx version.

We probably shouldn't merge this until Sphinx 4 gets a stable release, that way we don't need to pin ``4.0.0b1``

**Review by foster999 — APPROVED** [review]
Looks good to me 👍

Agree that holding on for a more stable release might be best. Sounds like it might only be the end of the month

File: setup.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def get_version():
     url="https://github.com/executablebooks/sphinx-tabs",
     license="MIT",
     python_requires="~=3.6",
-    install_requires=["sphinx>=2,<4", "pygments"],
+    install_requires=["sphinx>=2,<5", "pygments", "docutils<=0.16.0"],
```
Comment (pradyunsg) [inline setup.py:28]: Did you mean to say the following?

```suggestion
    install_requires=["sphinx>=2,<5", "pygments", "docutils~=0.16.0"],
```

Comment (Daltz333) [reply setup.py:28]: Wasn't sure if we wanted to allow lower versions of docutils or not. Probably fine.

**Comment by Daltz333** [comment]
Release is only 5 days away. I'm going to go ahead and merge it, and we can test Sphinx 4 major/minor versions in a later PR.

**Pull request merged**